include/matioCpp/StructArrayElement.h

```diff
@@ -145,6 +145,15 @@ class matioCpp::StructArrayElement
     template<bool B = isConst, typename = typename std::enable_if_t<(B == isConst) && !B>>
     bool setField(index_type index, const Variable& newValue) const;
 
+    /**
+     * @brief Set the field given the field name
+     * @param field The field to change
+     * @param newValue The Variable that will be copied in the specified field (the name is not considered)
+     * @return True if successful, false otherwise (for example if the newValue is not valid or the field is not found)
+     */
+    template<bool B = isConst, typename = typename std::enable_if_t<(B == isConst) && !B>>
+    bool setField(const std::string& field, const Variable& newValue) const;
```

Comment:
Could you elaborate on this SFINAE construct, just for my information?

Reply:
Sure. The ``StructArrayElement`` is supposed to be a ``visitor`` of a ``StructArray``, much like an ``iterator``, but without the common iterator operations (like ``++``). For this reason, it is important to keep track of whether the array it is visiting is ``const`` or not. In order to do so, there is a boolean template parameter to check if the original array is ``const`` or not. In case the original array is ``const``, it cannot be modified, hence the ``StructArrayElement`` should just allow accessing the data, not modifying it. An easy version of it would be
```
template<typename = std::enable_if_t<!isConst>>
```
The problem with this is that SFINAE works only with **deduced** template arguments. For this reason, there is an additional boolean template argument ``B``, that is set equal to ``isConst``. In principle, one may use this method by explicitly setting ``B`` to a different value, but I prevent this case with a static assert in the function body. 

An additional nitpick. This method is ``const``, because it does not change the state of the specific ``StructArrayElement``, but it changes the state of the ``StructArray`` from which it is obtained. For this reason, the template machinery is necessary.

Reply:
Thank you for the explanation. It was clear.

But do you see a way to reduce the SFINAE using [`std::is_const`](https://en.cppreference.com/w/cpp/types/is_const)? Or is it not possible?

Reply:
``is_const`` is used to check if the template argument is ``const``. In this case, it is the ``StructArrayElement`` that acts as ``const`` for the object that points to.

Reply:
Here are some aliases: 
https://github.com/dic-iit/matio-cpp/blob/15c72f8e5896e8279f4bd5e54ed3787b18802fcf/include/matioCpp/StructArray.h#L39-L41

and here it is how it used inside ``StructArray``
https://github.com/dic-iit/matio-cpp/blob/15c72f8e5896e8279f4bd5e54ed3787b18802fcf/include/matioCpp/StructArray.h#L256-L271

Notice that the ``const`` methods return a ``ConstElement``.

Reply:
Ah ok! That clarifies all of my doubts. Thanks for the explanations!! 
I will just need a bit more processing on my own.